app/actions/remote/channel.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {General} from '@constants';
+import DatabaseManager from '@database/manager';
+import NetworkManager from '@init/network_manager';
+import {prepareMyChannelsForTeam} from '@queries/servers/channel';
+import {displayGroupMessageName, displayUsername} from '@utils/user';
+
+import type {Model} from '@nozbe/watermelondb';
+
+import {forceLogoutIfNecessary} from './session';
+import {fetchProfilesPerChannels} from './user';
+
+export type MyChannelsRequest = {
+    channels?: Channel[];
+    memberships?: ChannelMembership[];
+    error?: never;
+}
+
+export const fetchMyChannelsForTeam = async (serverUrl: string, teamId: string, includeDeleted = true, since = 0, fetchOnly = false, excludeDirect = false): Promise<MyChannelsRequest> => {
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        let [channels, memberships] = await Promise.all<Channel[], ChannelMembership[]>([
+            client.getMyChannels(teamId, includeDeleted, since),
+            client.getMyChannelMembers(teamId),
+        ]);
+
+        if (excludeDirect) {
+            channels = channels.filter((c) => c.type !== General.GM_CHANNEL && c.type !== General.DM_CHANNEL);
+        }
+
+        const channelIds = new Set<string>(channels.map((c) => c.id));
+        memberships = memberships.reduce((result: ChannelMembership[], m: ChannelMembership) => {
+            if (channelIds.has(m.channel_id)) {
+                result.push(m);
+            }
+            return result;
+        }, []);
+
+        if (!fetchOnly) {
+            const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+            const modelPromises: Array<Promise<Model[]>> = [];
+            if (operator) {
```

Comment:
1. Can the operator for this serverUrl be undefined? 
2. Do we care?
3. What would happen to the caller/app if we were meant to write to DB  and we don't have an operator?

Reply:
1. Not really, as this function shouldn't be called before database are created, this is just to comply with TS instead of using the `!` operator
2. No but see 1
3. it won't write to the DB, now the question is what would call this function without having a db already set. If you follow the code, you'll see that the callers are already checking for db existence.